diff --git a/eatmemory.c b/eatmemory.c
index 481918d..2a88382 100644
--- a/eatmemory.c
+++ b/eatmemory.c
@@ -128,20 +128,10 @@ int main(int argc, char *argv[]){
     printf("Eating %ld bytes in chunks of %d...\n",size,chunk);
     short** eaten = eat(size,chunk);
     if(eaten){
-        if(timeout < 0 && isatty(fileno(stdin))) {
-            printf("Done, press ENTER to free the memory\n");
-            getchar();
-        } else if (timeout >= 0) {
-            printf("Done, sleeping for %d seconds before exiting...\n", timeout);
-            sleep(timeout);
-        } else {
-            printf("Done, kill this process to free the memory\n");
-            while(true) {
-                sleep(1);
-            }
-        }
-        digest(eaten, size, chunk);
-    }else{
+        printf("Done\n");
+        exit(0);
+    }
+    else{
         printf("ERROR: Could not allocate the memory");
     }
