diff --git a/eatmemory.c b/eatmemory.c
index e5891fa..8b1e345 100644
--- a/eatmemory.c
+++ b/eatmemory.c
@@ -86,16 +86,10 @@ int main(int argc, char *argv[]){
             }
             printf("Eating %ld bytes in chunks of %d...\n",size,chunk);
             if(eat(size,chunk)){
-                if(isatty(fileno(stdin))) {
-                    printf("Done, press any key to free the memory\n");
-                    getchar();
-                } else {
-                    printf("Done, kill this process to free the memory\n");
-                    while(true) {
-                        sleep(1);
-                    }
-                }
-            }else{
+                printf("Done\n");
+                exit(0);
+            }
+            else{
                 printf("ERROR: Could not allocate the memory");
             }
         }
